aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -254,6 +254,10 @@ public class DefaultModule : Module
 
 At runtime, Autofac will be used to resolve types and inject dependencies. [Learn more about using Autofac and ASP.NET Core](http://docs.autofac.org/en/latest/integration/aspnetcore.html).
 
+### Thread safety
+
+Singleton services need to be thread safe. If a singleton service has a dependency on a transient service, the  transient service may also need to be thread safe depending how it’s used by the singleton.
```

Comment:
depending how --> depending on how

This is quite spare ... it would be ideal if you could link to something for more information.  "depending on how it's used" begs the question of what way of using a dependency would make the dependency need to be thread-safe.